File: cpp/unittests/ImplementTriePrefixTreeTest.cpp
#include "ImplementTriePrefixTree.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ImplementTriePrefixTree, Solution208_1) {
  Trie T;
  T.insert("apple");
  EXPECT_TRUE(T.search("apple"));
  EXPECT_FALSE(T.search("app"));
  EXPECT_TRUE(T.startsWith("app"));
  T.insert("app");
  EXPECT_TRUE(T.search("app"));
}

File: java/src/test/java/io/binac/leetcode/ImplementTriePrefixTreeTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ImplementTriePrefixTreeTest {

    @Test
    void test1() {
        ImplementTriePrefixTree.Trie trie = new ImplementTriePrefixTree.Trie();
        trie.insert("apple");
        assertTrue(trie.search("apple"));
        assertFalse(trie.search("app"));
        assertTrue(trie.startsWith("app"));
        trie.insert("app");
        assertTrue(trie.search("app"));
    }
}